crates/trie/parallel/src/proof_task.rs

```diff
@@ -334,6 +414,56 @@ impl<Tx> ProofTaskManagerHandle<Tx> {
     }
 }
 
+impl<Tx: DbTx> BlindedProviderFactory for ProofTaskManagerHandle<Tx> {
+    type AccountNodeProvider = ProofTaskBlindedNodeProvider<Tx>;
+    type StorageNodeProvider = ProofTaskBlindedNodeProvider<Tx>;
+
+    fn account_node_provider(&self) -> Self::AccountNodeProvider {
+        ProofTaskBlindedNodeProvider::AccountNode { sender: self.sender.clone() }
+    }
+
+    fn storage_node_provider(&self, account: B256) -> Self::StorageNodeProvider {
+        ProofTaskBlindedNodeProvider::StorageNode { account, sender: self.sender.clone() }
+    }
+}
+
+/// Blinded node provider for retrieving trie nodes by path.
+#[derive(Debug)]
+pub enum ProofTaskBlindedNodeProvider<Tx> {
+    /// Blinded account trie node provider.
+    AccountNode {
+        /// Sender to the proof task.
+        sender: Sender<ProofTaskMessage<Tx>>,
+    },
+    /// Blinded storage trie node provider.
+    StorageNode {
+        /// Target account.
+        account: B256,
+        /// Sender to the proof task.
+        sender: Sender<ProofTaskMessage<Tx>>,
+    },
+}
+
+impl<Tx: DbTx> BlindedProvider for ProofTaskBlindedNodeProvider<Tx> {
+    fn blinded_node(&self, path: &Nibbles) -> Result<Option<RevealedNode>, SparseTrieError> {
+        let (tx, rx) = channel();
+        match self {
+            Self::AccountNode { sender: tx_sender } => {
+                let _ = tx_sender.send(ProofTaskMessage::QueueTask(
+                    ProofTaskKind::BlindedAccountNode(path.clone(), tx),
+                ));
+            }
+            Self::StorageNode { sender: tx_sender, account } => {
+                let _ = tx_sender.send(ProofTaskMessage::QueueTask(
+                    ProofTaskKind::BlindedStorageNode(*account, path.clone(), tx),
+                ));
+            }
```

Comment:
these clones aren't good and should be avoided

Reply:
yeah... I think we can solve it with https://github.com/paradigmxyz/reth/pull/14598 so an Arc is passed here